File: objc/Project3/Practice1/Views/WindowController.m
//
//  WindowController.m
//  Practice1
//
//  Created by Jinwoo Kim on 11/22/20.
//

#import "WindowController.h"
#import "DetailViewController.h"

@interface WindowController ()
@property (strong) IBOutlet NSButton *shareButton;
@end

@implementation WindowController

- (void)windowDidLoad {
    [super windowDidLoad];
    [self.shareButton sendActionOn:NSEventMaskLeftMouseDown];
}

- (IBAction)shareClicked:(NSView *)sender {
    NSSplitViewController *split = (NSSplitViewController *)self.contentViewController;
    DetailViewController *detail = split.childViewControllers[1];
    NSImage *image = detail.imageView.image;
    
    NSSharingServicePicker *picker = [[NSSharingServicePicker alloc] initWithItems:@[image]];
    [picker showRelativeToRect:CGRectZero ofView:sender preferredEdge:NSRectEdgeMinY];
}


@end

File: swift/Project3/Project1/Views/WindowController.swift
//
//  WindowController.swift
//  Project1
//
//  Created by Jinwoo Kim on 11/22/20.
//

import Cocoa

class WindowController: NSWindowController {
    @IBOutlet var shareButton: NSButton!
    
    override func windowDidLoad() {
        super.windowDidLoad()
        shareButton.sendAction(on: .leftMouseDown)
    }
    
    @IBAction func shareClicked(_ sender: NSView) {
        guard let split = contentViewController as? NSSplitViewController else { return }
        guard let detail = split.children[1] as? DetailViewController else { return }
        guard let image = detail.imageView.image else { return }
        
        let picker = NSSharingServicePicker(items: [image])
        picker.show(relativeTo: .zero, of: sender, preferredEdge: .minY)
    }
}
